api/placer.py

```diff
@@ -138,7 +138,8 @@ def check(self):
         metadata_validator = validators.from_schema_path(payload_schema_uri)
         metadata_validator(self.metadata, 'POST')
 
-        targets = self.create_hierarchy(self.metadata)
+        # If not a superuser request, pass uid of user making the upload request
```

Comment:
I think it would be possible to check here user permissions.
This would keep permissions checking separate from database access.